test/screenshot/tab-bar/index.js

```diff
@@ -9,6 +9,12 @@ class TabBarTest extends React.Component {
     activeIndex: this.props.activeIndex || 0, // eslint-disable-line react/prop-types
   };
 
+  componentDidUpdate(prevProps) {
+    if (this.props.activeIndex !== prevProps.activeIndex) {
+      this.setState({activeIndex: this.props.activeIndex});
+    }
+  }
+
   render() {
     const {
```

Comment:
```suggestion
    const {
      /* eslint-disable react/prop-types */
      className,
      indicatorContent,
      isFadingIndicator,
      numTabs,
      isRtl,
      hasUpdateActiveIndexButton,
      /* eslint-enable react/prop-types */
    } = this.props;

    const renderTab = (num, index) => {
      return (
        <Tab
          key={index}
          isFadingIndicator={isFadingIndicator}
          indicatorContent={indicatorContent}
        >
          <span className='mdc-tab__text-label'>Tab {num + 1}</span>
        </Tab>
      );
    };

    return (
      <React.Fragment>
        <TabBar
          isRtl={isRtl}
          className={className}
          activeIndex={this.state.activeIndex}
          handleActiveIndexUpdate={(activeIndex) => this.setState({activeIndex})}
        >
          {[...Array(numTabs).keys()].map(renderTab)}
        </TabBar>
        {hasUpdateActiveIndexButton ? <button onClick={() => this.setState({activeIndex: 1})}>Click</button> : null}
      </React.Fragment>
    );
```

Reply:
I think we're trying to wrap too many layers, which makes the screenshot test code very error prone. How about we add the button inside `TabBarTest` instead and hide it behind a boolean like `hasUpdateActiveIndexButton`?